src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerEditorPanel.java

```diff
@@ -0,0 +1,235 @@
+      public void actionPerformed(ActionEvent e) {
+        HttpConfigurable.editConfigurable(p);
+        enableProxy.setEnabled(HttpConfigurable.getInstance().USE_HTTP_PROXY);
+        apply();
+      }
+    });
+
+    installListener(nameText);
+    installListener(urlText);
+    installListener(loginText);
+    installListener(passwordText);
+    installListener(enableProxy);
+    installListener(idText);
+
+    return p;
+  }
+
+  private void addForm(JPanel p) {
+    JPanel form = new CustomFormBuilder()
+      .setAlignLabelOnRight(true)
+      .setFormLeftIndent(5)
+      .addLabeledComponent(nameLabel, nameText)
+      .addLabeledComponent(idLabel, idText)
+      .addLabeledComponent(urlLabel, urlText)
+      .addLabeledComponent(loginLabel, loginText)
+      .addLabeledComponent(passwordLabel, passwordText)
+      .addVerticalGap(5)
+      .addSeparator()
+      .addLabeledComponent(enableProxy, proxySettingsButton)
+      .getPanel();
+
+    GridBagConstraints gbc = new GridBagConstraints();
+    gbc.anchor = GridBagConstraints.NORTH;
+    gbc.fill = GridBagConstraints.HORIZONTAL;
+    gbc.gridy = 0;
+    gbc.gridx = 0;
+    gbc.weightx = 1.0;
+    p.add(form, gbc);
+
+    gbc = new GridBagConstraints();
+    gbc.anchor = GridBagConstraints.NORTHEAST;
+    gbc.gridy = 1;
+    gbc.gridx = 0;
+    gbc.weighty = 1.0;
+    p.add(testButton, gbc);
+  }
+
+  private void testConnection() {
+    ConnectionTestTask test = new ConnectionTestTask(server);
+    ProgressManager.getInstance().run(test);
+    ValidationResult r = test.result();
+
+    if(test.getException() != null) {
+      Messages.showErrorDialog(testButton, "Error testing connection: " + test.getException().getMessage(), "Error");
+    } else if(r.status()) {
+      Messages.showMessageDialog(testButton, r.message(), "Connection", Messages.getInformationIcon());
+    } else {
+      Messages.showErrorDialog(testButton, r.message(), "Connection failed");
+    }
+  }
+
+  protected void installListener(JTextField textField) {
+    textField.getDocument().addDocumentListener(new DocumentAdapter() {
+      @Override
+      protected void textChanged(DocumentEvent e) {
+        apply();
+      }
+    });
+  }
+
+  protected void installListener(JBCheckBox checkBox) {
+    checkBox.addChangeListener(new ChangeListener() {
+      @Override public void stateChanged(ChangeEvent e) {
+        apply();
+      }
+    });
+  }
+
+  void apply() {
+    server.setName(nameText.getText().trim());
+    server.setServerId(idText.getText().trim());
+    server.setHostUrl(urlText.getText().trim());
+    server.setLogin(loginText.getText().trim());
+    server.setPassword(new String(passwordText.getPassword()));
+    server.setEnableProxy(enableProxy.isSelected());
+
+    changeListener.consume(server);
+  }
+
+  @Override public void dispose() {
+  }
+
+  /**
+   * A {@link FormBuilder} that doesn't fill buttons
+   */
+  private class CustomFormBuilder extends FormBuilder {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)